Sources/FluentKit/Query/QueryBuilder+Unique.swift

```diff
@@ -0,0 +1,16 @@
+extension QueryBuilder {
+    public func all<Result>(uniqueOn key: KeyPath<Model, Field<Result>>) -> EventLoopFuture<[Result]> where Result: Decodable {
+        let copy = self.copy()
+        let fieldKey = Model.key(for: key)
+        copy.query.isUnique = true
+        copy.query.fields = [.field(path: [fieldKey], schema: Model.schema, alias: nil)]
+        return copy.all().flatMapThrowing { models in
+            return try models.map { try $0._$id.cachedOutput!.decode(fieldKey, as: Result.self) }
```

Comment:
We should be able to return `model[keyPath: key].wrappedValue` here.

Reply:
Updated 👍